From 8301e01a16363ddb7f68e9e4dc60abb50a6cf819 Mon Sep 17 00:00:00 2001
From: Stefan Roese <sr@denx.de>
Date: Mon, 17 May 2021 11:44:29 +0200
Subject: [PATCH 9/9] net: DSA: mt7628-esw: Don't return with -ENODEV in
 port_enable() for port 0

Returning with -ENODEV in port_enable leaves the virtual network device
disabled. This results in inaccessablilty of some SYSFS entries, like
"carrier", checking netif_running() before returning the value.

This makes it possible to use the network interface "lan" (just an
example) which is has the cable connected (or not) for such SYSFS
queries:

cat /sys/class/net/lan/carrier

Signed-off-by: Stefan Roese <sr@denx.de>
---
 drivers/net/dsa/mt7628-esw.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/dsa/mt7628-esw.c b/drivers/net/dsa/mt7628-esw.c
index ba0e368074ff..43e7fcc70ca1 100644
--- a/drivers/net/dsa/mt7628-esw.c
+++ b/drivers/net/dsa/mt7628-esw.c
@@ -684,9 +684,9 @@ static int mt7628_esw_port_enable(struct dsa_switch *ds, int port,
 	struct mt7628_esw_priv *priv = ds->priv;
 
 //	printk("%s (%d): port=%d\n", __func__, __LINE__, port); // test-only
-	/* Skip port 0 enabling */
+	/* Skip port 0 enabling, just return with success */
 	if (port == 0)
-		return -ENODEV;
+		return 0;
 
 	priv->ports[port].enable = true;
 
-- 
2.29.2

